process/block/metablock.go

```diff
@@ -1223,7 +1224,11 @@ func (mp *metaProcessor) CommitBlock(
 		mp.blockTracker.CleanupInvalidCrossHeaders(header.Epoch, header.Round)
 	}
 
-	mp.indexBlock(header, headerHash, body, lastMetaBlock, notarizedHeadersHashes, rewardsTxs)
+	err = mp.indexBlock(header, headerHash, body, lastMetaBlock, notarizedHeadersHashes, rewardsTxs)
+	if err != nil {
+		return err
```

Comment:
Is this critical? We should return here? If yes, this return will block the node as the commit has been already done above. Also the set of current block header and current block hash in blockchain component is already done. I would move this line as upper it is possible, although this will not solve the blocking situation until we can move this call above L1161.